// storable-react/test4.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _storablejsInventory = require("storablejs/Inventory");

var _storablejsInventory2 = _interopRequireDefault(_storablejsInventory);

var _storablejsStore = require("storablejs/Store");

var _storablejsStore2 = _interopRequireDefault(_storablejsStore);

var _storablejsManager = require("storablejs/Manager");

var _storablejsManager2 = _interopRequireDefault(_storablejsManager);

var _Connect = require("./Connect");

var _Connect2 = _interopRequireDefault(_Connect);

var _StoreDebugger = require("./StoreDebugger");

var _StoreDebugger2 = _interopRequireDefault(_StoreDebugger);

var SubApp = _react2["default"].createClass({
    displayName: "SubApp",

    render: function render() {
        return _react2["default"].createElement(
            "div",
            null,
            _react2["default"].createElement(
                "div",
                null,
                this.props.hoo
            ),
            _react2["default"].createElement(
                "div",
                null,
                this.props.dooky
            )
        );
    }
});

exports["default"] = _react2["default"].createClass({
    displayName: "test3",

    render: function render() {
        var i = new _storablejsInventory2["default"]();
        var m = new _storablejsManager2["default"](i);
        var s = new _storablejsStore2["default"](i, m);

        i.contents.forEach(function (state) {
            console.log("STATE:", state);
        });

        function TestModel2() {
            this.thing = function () {};
        }

        function TestModel() {
            this.a = new TestModel2();
            this.b = 123;
            this.c = "This is a string";
            this.d = true;
            this.e = {};
            this.f = [1, 2, 3];
        };

        window.doNext = function () {
            s.emit({
                foo: { bar: Math.ceil(Math.random() * 100) },
                baz: "heelo",
                hoo: new TestModel()
            });
        };

        return _react2["default"].createElement(
            "div",
            null,
            _react2["default"].createElement(
                _Connect2["default"],
                { store: s,
                    keyPaths: [["foo", "bar", { alias: "hoo" }], ["hoo", "f", { alias: "dooky" }]] },
                _react2["default"].createElement(SubApp, null)
            ),
            _react2["default"].createElement(_StoreDebugger2["default"], { manager: m })
        );
    }
});
module.exports = exports["default"];
